src/Lean/Elab/Coinductive.lean

```diff
@@ -0,0 +1,217 @@
+/-
+Copyright (c) 2020 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: William Sørensen
+-/
+prelude
+import Lean.Elab.Command
+import Lean.Elab.DeclarationRange
+import Lean.Elab.Exception
+import Lean.Elab.Inductive
+import Lean.Util.Trace
+
+namespace Lean.Elab.Command
+
+open Lean Lean.Elab Lean.Elab.Term Lean.Parser Command
+
+builtin_initialize
+  registerTraceClass `Elab.CoInductive
+
+structure CoInductiveView.CtorView where
+  ref       : Syntax
+  modifiers : Modifiers
+  declName  : Name
+  binders   : TSyntaxArray ``Lean.Parser.Term.bracketedBinder
+  type?     : Option Term
+  deriving Inhabited
```

Comment:
The reason I defined this is because the original uses `Syntax` instead of `TSyntax` at multiple places. I can change over to theirs and add casts and stuff if you'd prefer